stats/registry_test.go

```diff
@@ -0,0 +1,50 @@
+package stats_test
+
+import (
+	"testing"
+
+	"github.com/influxdata/influxdb/stats"
+)
+
+func TestEmptyStatistics(t *testing.T) {
+	found := stats.Collect(stats.Root.Open(), true)
+
+	if length := len(found); length != 0 {
+		t.Fatalf("non empty initial state. got %d, expected: %d", length, 0)
+	}
+}
+
+// Test that we can create one statistic and that it disappears after it is deleted twice.
+func TestOneStatistic(t *testing.T) {
+
```

Comment:
Of course - GC programming languages 101.

However, the intent of the foo = nil assignment was really to force the execution of a code path in:

```
defer func() {
    if foo != nil {
        foo.Close()
    }
}
```

In your initial review comment you appeared to think that I could I have issued:

```
defer foo.Close() 
```

at line 26, and deleted line 42.

However, this would have defeated the point of the test where I needed foo.Close() to be called before as test I wanted to make after foo.Close() had been called.

The current code now which now explicitly scopes the lifetime of foo to a closure, uses defer foo.Close() in an idiomatic way and allows me to do the test I wanted to do. It might helpful, I think to look at the current code in GitHub to understand the point I am trying to make.